crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -56,6 +56,83 @@ impl SystemMeta {
 // TODO: Actually use this in FunctionSystem. We should probably only do this once Systems are constructed using a World reference
 // (to avoid the need for unwrapping to retrieve SystemMeta)
 /// Holds on to persistent state required to drive [`SystemParam`] for a [`System`].
+///
+/// This is a very powerful and convenient tool for working with exclusive world access,
+/// allowing you to fetch data from the [`World`] as if you were running a [`System`].
+///
+/// Borrow-checking is handled for you, allowing you to mutably access multiple compatible system parameters at once,
+/// and arbitrary system parameters (like [`EventWriter`](crate::event::EventWriter)) can be conveniently fetched.
+///
+/// For an alternative approach to split mutable access to the world, see [`World::resource_scope`].
+///
+/// # Warning
+/// This API has more performance overhead than more direct methods: prefer them when possible.
```

Comment:
Is this warning really necessary? I can't think of a case where someone would use this api over more direct ones unless they need the extra flexibility.